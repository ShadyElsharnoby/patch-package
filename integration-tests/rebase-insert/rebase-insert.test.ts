import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "rebase-insert",
  shouldProduceSnapshots: true,
})
