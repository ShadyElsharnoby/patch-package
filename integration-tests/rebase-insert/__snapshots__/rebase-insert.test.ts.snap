// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rebase-insert: 00: Rebase to the second patch 1`] = `
"SNAPSHOT: Rebase to the second patch
patch-package 0.0.0
Un-applied left-pad+1.3.0+003+goodbye.patch

Make any changes you need inside node_modules/left-pad

When you are done, do one of the following:

  To update left-pad+1.3.0+002+world.patch run

    patch-package left-pad
    
  To create a new patch file after left-pad+1.3.0+002+world.patch run
  
    patch-package left-pad --append 'MyChangeDescription'

  
END SNAPSHOT"
`;

exports[`Test rebase-insert: 01: the state file should show two patches applied and isRebasing: true 1`] = `
"SNAPSHOT: the state file should show two patches applied and isRebasing: true
{
    \\"isRebasing\\": true,
    \\"patches\\": [
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"404c604ed830db6a0605f86cb9165ced136848f70986b23bf877bcf38968c1c9\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+001+hello.patch\\"
        },
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"f2859c7193de8d9578bdde7e226de516adc8d972d6e76997cbe1f41b1a535359\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+002+world.patch\\"
        }
    ],
    \\"version\\": 1
}END SNAPSHOT"
`;

exports[`Test rebase-insert: 02: insert a patch in the sequence and fast forward to the end 1`] = `
"SNAPSHOT: insert a patch in the sequence and fast forward to the end
patch-package 0.0.0
• Creating temporary folder
• Installing left-pad@1.3.0 with npm
• Diffing your files with clean files
Renaming left-pad+1.3.0+003+goodbye.patch to left-pad+1.3.0+004+goodbye.patch
✔ Created file patches/left-pad+1.3.0+003+some-stuff.patch

Fast forwarding...
  ✔ left-pad+1.3.0+004+goodbye.patch
ls patches
left-pad+1.3.0+001+hello.patch
left-pad+1.3.0+002+world.patch
left-pad+1.3.0+003+some-stuff.patch
left-pad+1.3.0+004+goodbye.patch
END SNAPSHOT"
`;

exports[`Test rebase-insert: 03: the state file should show three patches applied and isRebasing: false 1`] = `
"SNAPSHOT: the state file should show three patches applied and isRebasing: false
{
    \\"isRebasing\\": false,
    \\"patches\\": [
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"404c604ed830db6a0605f86cb9165ced136848f70986b23bf877bcf38968c1c9\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+001+hello.patch\\"
        },
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"f2859c7193de8d9578bdde7e226de516adc8d972d6e76997cbe1f41b1a535359\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+002+world.patch\\"
        },
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"bcfdda02dae96c0ebc805877bf6497cc77ba50c1f1a84b9dab9c2b9ffcfa6fbe\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+003+some-stuff.patch\\"
        },
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"c9063ed6ae00867ee243fa71590c369ce0bb699f3a63a10df86d3ec988782715\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+004+goodbye.patch\\"
        }
    ],
    \\"version\\": 1
}END SNAPSHOT"
`;

exports[`Test rebase-insert: 04: The patch file created only shows the new bits 1`] = `
"SNAPSHOT: The patch file created only shows the new bits
diff --git a/node_modules/left-pad/index.js b/node_modules/left-pad/index.js
index 5aa41be..8b88742 100644
--- a/node_modules/left-pad/index.js
+++ b/node_modules/left-pad/index.js
@@ -50,3 +50,4 @@ function leftPad (str, len, ch) {
   // pad \`str\`!
   return pad + str;
 }
+'some stuff'
END SNAPSHOT"
`;
